CMakeLists.txt

```diff
@@ -32,6 +32,11 @@ option(ENABLE_CLANG_TIDY_FIX   "Analyze code with clang-tidy and fix errors."
 option(ENABLE_COVERAGE         "Add coverage information to binaries."               OFF)
 option(ENABLE_COLOR_OUTPUT     "Force produce ANSI-colored output (GNU/Clang only)."  ON)
 option(ENABLE_LTO              "Enable link-time-optimization. Increases link time." OFF)
+option(USE_EXTERN_QUAZIP       "Enable link-time-optimization. Increases link time." OFF)
+
+if(USE_EXTERN_QUAZIP)
+  add_definitions(-DEXTERN_QUAZIP)
```

Comment:
I try to avoid non-target-specific functions. I would prefer to move this to the `mediaelch_post_target_defaults` function that is applied to each of MediaElch's targets :)

Reply:
This is that 'modern CMake' thing, isn't it?
Bit of a grey area for me, personally. But after your comment and looking at the CMakeLists.txt, I see your point - the define setting should be in the function(~line 60).
I can rework and resubmit later. My mess to clean this up.

Reply:
Yeah, exactly! "Modern" CMake is what I try to use (we require CMake v3.13 after all)
Don't worry, I'll push a few changes in a few minutes (after I successfully compiled MediaElch with my system's QuaZip library) :)

Reply:
I had to update my CMake to v3.15 to work this(was v3.12). It's kinda of cool reading some of the theory. I get the premise. Implementation and general practice is an entirely different. Good to have a working example as a reference.

Reply:
Well, I try to have this as modern as it can be. Though there are a few places where we may not exactly be "best practice". e.g. we use `mediaelch_post_target_defaults` which I think isn't best practice. But it works... :)

Reply:
But makes phenomenal sense in hindsight. I can see the utility here.